### ISSUE TITLE
Error, relator wrong - fresh problems in sanity.g

### ISSUE THREAD

**Issue by alex-konovalov** [gap-system/gap#3517]
New problems observed in sanity.g this weekend. GAP is started with default packages, with `../bin/gap.sh --quitonbreak sanity.g`. 

1) In the master branch, on 64-bit Linux:
```
testing group 1/2[2^6]L(6):2
normal subgroups
Error, relator wrong at /circa/scratch/gap-jenkins/workspace/GAP-master-sanity/GAPCOPTS/64build/la\
bel/kovacs/GAP-master-snapshot/lib/onecohom.gi:679 called from
```
while in 32-bit Linux and both 32,64-bit macOS builds:
```
testing group t16n1653
normal subgroups
Error, relator wrong at /circa/scratch/gap-jenkins/workspace/GAP-master-sanity/GAPCOPTS/32build/la\
bel/kovacs/GAP-master-snapshot/lib/onecohom.gi:679 called from
```
See [St Andrews only link to the build](https://gap-ci.cs.st-andrews.ac.uk/view/GAP-weekly/job/GAP-master-sanity/11/).

2. In the stable-4.10 branch, on Linux:
```
testing group t18n935
conjugacy classes
Error, Assertion failure in
  Assert( 2, Sum( clout, function ( i )
          return Index( F, i[2] );
      end ) = Size( F ) - Size( M ) 
 ); at /circa/scratch/gap-jenkins/workspace/GAP-stable-sanity/GAPCOPTS/32build\
/label/kovacs/GAP-stable-snapshot/lib/clashom.gi:1050 called from 
```
in both 32 and 64-bit. Perhaps on macOS too, but the test timeouts slightly earlier, as that machine is slower.

See [St Andrews only link to the build](https://gap-ci.cs.st-andrews.ac.uk/view/GAP-weekly/job/GAP-stable-sanity/41/).

**Comment by hulpke** [comment]
Which version are the first two bugs on? Development or 4.10? What is the last known time this test ran through? Do the errors arise if one starts the sanity test just before the groups ? I.e. Do

    Sanity(12,255,80,1);
    Sanity(16,1650,80,1);
    Sanity(18,934,80,1);

run into an error quickly? (they do not for me).

With the errors happening at different places in different architectures, the only reason I could imagine is the random number generator (no idea whether it is 32/64 bit dependent) or memory issues. (Or kernel bugs that one compiler does not expose.)

**Comment by alex-konovalov** [comment]
Problem 1 disappeared.

Problem 2 is occurring in stable-4.10 branch:
```
vpn-10:gap-stable-4.10 alexk$ bin/gap.sh -r -A tst/sanity.g 
 ┌───────┐   GAP 4.10.2-1-g2dbf56e of today
 │  GAP  │   https://www.gap-system.org
 └───────┘   Architecture: x86_64-apple-darwin17.7.0-default64-kv3
 Configuration:  gmp 6.1.2, readline
 Loading the library and packages ...
 Packages:   GAPDoc 1.6.2, PrimGrp 3.3.2, SmallGrp 1.3, TransGrp 2.0.4
 Try '??help' for help. See also '?copyright', '?cite' and '?authors'
gap> for i in [1..100] do Print(i,"\n");TestsForGroup(TransitiveGroup(18,935)); od;
1
testing group t18n935
conjugacy classes
rational classes
automorphism group
2
testing group t18n935
conjugacy classes
rational classes
automorphism group
3
testing group t18n935
conjugacy classes
rational classes
automorphism group
4
testing group t18n935
conjugacy classes
rational classes
automorphism group
5
testing group t18n935
conjugacy classes
rational classes
automorphism group
6
testing group t18n935
conjugacy classes
rational classes
automorphism group
7
testing group t18n935
conjugacy classes
rational classes
automorphism group
8
testing group t18n935
conjugacy classes
Error, Assertion failure in
  Assert( 2, Sum( clout, function ( i )
          return Index( F, i[2] );
      end ) = Size( F ) - Size( M ) 
 ); at /Users/alexk/GITREPS/gap-stable-4.10/lib/clashom.gi:1050 called from 
ConjugacyClassesSubwreath( F, FM, n, S1, Action( FM, components[1] ), T1, 
 components, emb, proj 
 ) at /Users/alexk/GITREPS/gap-stable-4.10/lib/clashom.gi:1339 called from
ConjugacyClassesFittingFreeGroup( G 
 ) at /Users/alexk/GITREPS/gap-stable-4.10/lib/clashom.gi:2363 called from
ConjugacyClassesViaRadical( G 
 ) at /Users/alexk/GITREPS/gap-stable-4.10/lib/clashom.gi:3078 called from
ConjugacyClasses( G ) at tst/sanity.g:61 called from
TestsForGroup( TransitiveGroup( 18, 935 ) ); at *stdin*:1 called from
...  at *stdin*:1
you may 'return;'
brk> 
```

**Comment by fingolfin** [comment]
I can reproduce this albeit not in step 8 but in step 37:
```
$ gap -A -r tst/sanity.g
 ┌───────┐   GAP 4.10dev-2199-gbf6501b of today
 │  GAP  │   https://www.gap-system.org
 └───────┘   Architecture: x86_64-apple-darwin18.7.0-default64-kv7
 Configuration:  gmp 6.1.2, GASMAN, readline
 Loading the library and packages ...
 Packages:   GAPDoc 1.dev, PrimGrp 3.3.2, SmallGrp 1.3, TransGrp 2.0.4
 Try '??help' for help. See also '?copyright', '?cite' and '?authors'
gap> for i in [1..100] do Print(i,"\n");TestsForGroup(TransitiveGroup(18,935)); od;
1
testing group t18n935
conjugacy classes
rational classes
automorphism group
2
testing group t18n935
conjugacy classes
rational classes
automorphism group
3
testing group t18n935
conjugacy classes
rational classes
automorphism group
4
testing group t18n935
conjugacy classes
rational classes
automorphism group
5
testing group t18n935
conjugacy classes
rational classes
automorphism group
6
testing group t18n935
conjugacy classes
rational classes
automorphism group
7
testing group t18n935
conjugacy classes
rational classes
automorphism group
8
testing group t18n935
conjugacy classes
rational classes
automorphism group
9
testing group t18n935
conjugacy classes
rational classes
automorphism group
10
testing group t18n935
conjugacy classes
rational classes
automorphism group
...
37
testing group t18n935
conjugacy classes
Error, Assertion failure in
  Sum( clout, function ( i )
        return Index( F, i[2] );
    end ) = Size( F ) - Size( M ) at /Users/mhorn/Projekte/GAP/gap.github/lib/clashom.gi:1057 called from
ConjugacyClassesSubwreath( F, FM, n, S1, Action( FM, components[1] ), T1, components, emb, proj ) at /Users/mhorn/Projekte/GAP/gap.github/lib/clashom.gi:1346 called from
ConjugacyClassesFittingFreeGroup( G ) at /Users/mhorn/Projekte/GAP/gap.github/lib/clashom.gi:2370 called from
ConjugacyClassesViaRadical( G ) at /Users/mhorn/Projekte/GAP/gap.github/lib/clashom.gi:3087 called from
ConjugacyClasses( G ) at tst/sanity.g:63 called from
TestsForGroup( TransitiveGroup( 18, 935 ) ); at *stdin*:1 called from
...  at *stdin*:1
you may 'return;'
```

**Comment by fingolfin** [comment]
@hulpke I still can reproduce this consistently in step 37 (see above)

**Comment by alex-konovalov** [comment]
Should be closed by #3704 - thanks @hulpke
